### ISSUE TITLE
credit card fraud detection

### ISSUE THREAD

**Issue by 11249a250-cloud** [11249a250-cloud/labmentix-#1]
GOOGLE COLAB LINK:https://colab.research.google.com/drive/1eroaRLGoKT5L2ryYbi0Aa9qVbPUFHcUc?usp=sharing
code:
# CREDIT CARD FRAUD DETECTION USING RANDOM FOREST

# Import required libraries
import pandas as pd
import zipfile
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# Extract ZIP file
with zipfile.ZipFile('/content/archive (3).zip', 'r') as zip_ref:
    zip_ref.extractall('/content/dataset')

print("ZIP Extracted Successfully!")

# Load the training dataset
df = pd.read_csv('/content/dataset/fraudTrain.csv')

print("Original Dataset Shape:", df.shape)

# Take a sample for faster training
df = df.sample(n=100000, random_state=42)

print("Sampled Dataset Shape:", df.shape)

# Remove unnecessary columns
df = df.drop([
    'Unnamed: 0',
    'trans_date_trans_time',
    'cc_num',
    'first',
    'last',
    'street',
    'city',
    'state',
    'dob',
    'trans_num'
], axis=1)

# Convert categorical columns into numerical values
for col in df.columns:
    if df[col].dtype == 'object':
        le = LabelEncoder()
        df[col] = le.fit_transform(df[col].astype(str))

# Handle missing values
df = df.fillna(0)

# Separate features and target variable
X = df.drop('is_fraud', axis=1)
y = df['is_fraud']

# Split dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)

# Create Random Forest model
model = RandomForestClassifier(
    n_estimators=20,
    random_state=42,
    n_jobs=-1
)

# Train the model
model.fit(X_train, y_train)

print("Model Training Completed!")

# Make predictions
y_pred = model.predict(X_test)

# Calculate accuracy
accuracy = accuracy_score(y_test, y_pred)

print("\nAccuracy:", accuracy)

# Display classification report
print("\nClassification Report:")
print(classification_report(y_test, y_pred))

# Display confusion matrix
print("\nConfusion Matrix:")
print(confusion_matrix(y_test, y_pred))

# Test with one sample transaction
sample_transaction = X_test.iloc[[0]]

prediction = model.predict(sample_transaction)

print("\nCustom Prediction:")

if prediction[0] == 1:
    print("Fraudulent Transaction")
else:
    print("Legitimate Transaction")
OUTPUT:
ZIP Extracted Successfully!
Original Dataset Shape: (1296675, 23)
Sampled Dataset Shape: (100000, 23)
Model Training Completed!

Accuracy: 0.99575

Classification Report:
              precision    recall  f1-score   support

           0       1.00      1.00      1.00     19884
           1       0.79      0.36      0.50       116

    accuracy                           1.00     20000
   macro avg       0.89      0.68      0.75     20000
weighted avg       1.00      1.00      0.99     20000

Confusion Matrix:
[[19873    11]
 [   74    42]]

Custom Prediction:
Legitimate Transaction
